DataciteExportDeployment.inc.php

```diff
@@ -21,294 +21,361 @@
+		$locale = ( $isSubmission === TRUE ) ? $object->getData( 'locale' ) : $parent->getData( 'locale' );
+		$creators = $documentNode->createElement( 'creators' );
+		if( $isSubmission === TRUE )
+		{
+			/** @var Publication $authors */
+			$authors = $object->getData('authors');
+			foreach( $authors as $author )
+			{
+				$creator = $this->createAuthor( $documentNode, $author, $locale );
+				if( $creator )
+				{
+					$creators->appendChild( $creator );
+					$documentNode->documentElement->appendChild( $creators );
 				}
 			}
-		} else {
-			$chapterAuthorDao = DAORegistry::getDAO('ChapterAuthorDAO');
-			$chapterAuthors = $chapterAuthorDao->getAuthors($object->getMonographId(), $object->getId());
-			while ($author = $chapterAuthors->next()) {
-				$creator = $this->createAuthor($documentNode, $author, $locale);
-				if ($creator) {
-					$creators->appendChild($creator);
-					$documentNode->documentElement->appendChild($creators);
+		}
+		else
+		{
+			/** @var Chapter $object */
+			$chapterAuthors = $object->getAuthors();
+			try
+			{
+				while( $author = $chapterAuthors->next() )
+				{
+					$creator = $this->createAuthor( $documentNode, $author, $locale );
+					if( $creator )
+					{
+						$creators->appendChild( $creator );
+						$documentNode->documentElement->appendChild( $creators );
+					}
 				}
-
 			}
-
-
+			catch( Exception $e )
+			{
+				DataciteExportPlugin::writeLog( $e, 'ERROR' );
+			}
 		}
 
 		return $documentNode;
 	}
 
-	function createAuthor($documentNode, $author, $locale)
+	/** @noinspection ReturnTypeCanBeDeclaredInspection */
+	private function createAuthor( $documentNode, $author, $locale )
 	{
-
-		$creator = $documentNode->createElement("creator");
-		$person = $documentNode->createElement("person");
-		$familyName = $author->getFamilyName($locale);
-		$givenName = $author->getGivenName($locale);
-		if ($familyName == '' && $givenName == '') {
-			return null;
+		$creator = $documentNode->createElement( 'creator' );
+		$person = $documentNode->createElement( 'person' );
+		$familyName = $author->getFamilyName( $locale );
+		$givenName = $author->getGivenName( $locale );
+
+		if( ( NULL === $familyName || empty( $familyName ) )
+			&& ( NULL === $givenName || empty( $givenName ) ) )
+		{
+			return NULL;
 		}
-		if ($this->getPlugin()->isDara()) {
-			if ($familyName != '') {
-				$lastName = $documentNode->createElement("lastName", $familyName);
-				$firstName = $documentNode->createElement("firstName", $givenName);
-				$person->appendChild($firstName);
-				$person->appendChild($lastName);
-				$creator->appendChild($person);
-			} else {
-				$institution = $documentNode->createElement("institution");
-				$institutionName = $documentNode->createElement("institutionName", $this->xmlEscape($givenName));
-				$institution->appendChild($institutionName);
-				$creator->appendChild($institution);
+
+		if ($this->getPlugin()->isDara())
+		{
```

Comment:
@nongenti 
ich habe dein Code angeschaut. Sieht sehr gut aus. Eine kleine Sache. In OPS/OMP wir trennen die funktionen mit {  in der gleichen Linie. Könntest Du bitte dein Editor so einstellen, dass die Funktionen so formatiert sind. Dann ist für mich das Vergleich viel leichter.

Reply:
@withanage
Ist erledigt. Wo finde ich denn Coding Conventions von PKP? Es gibt ja sicherlich noch mehr und würde meine IDE für die Zukunft gerne richtig einstellen.